stake-pool/program/src/processor.rs

```diff
@@ -1396,8 +1396,7 @@ impl Processor {
                 Some(stake_program::StakeState::Stake(meta, stake)) => {
                     let account_stake = meta
                         .rent_exempt_reserve
-                        .checked_add(stake.delegation.stake)
-                        .ok_or(StakePoolError::CalculationFailure)?;
+                        .saturating_add(stake.delegation.stake);
```

Comment:
Is this safe?  Seems like there must be some other bad math if we were throwing that error.  There shouldn't be enough Lamports globally to overflow the add :thinking:

Reply:
That's right, we should never overflow this -- the `checked` was only there out of an abundance of caution.  I didn't do a normal `+` to save you the trouble of having more crates to move away from unchecked math.

Reply:
This seems safe to me. However, nbd to wait and see if @dmakarov turns up something around the access violation that obviates this change.

Reply:
Works for me, I'll leave this open

Reply:
I think it's ok to merge.  There doesn't seem to be a code generation bug, but the stack size is over the limit in `spl_stake_pool::processor::Processor::process` (which becomes a big function with much on-stack data after many inner calls are being inlined). If this change helps to reduce the stack size somehow, and it's safe to make, by all means, please go ahead with merging this PR.